### PR TITLE
add integer subroutines to test_mpp_sendrecv.F90

### PR DESCRIPTION
**Pull request by wrongkindofdoctor**
**Description**
added subroutines to test mpp_send/mpp_recv with 2D and 3D 32-bit and 64-bit integer arrays to test_mpp_sendrecv.F90

**How Has This Been Tested?**
tested on AMD w/gcc 9 and intel20
tested on skylake w/gcc9, intel18up4, intel19
**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: test_fms/mpp/test_mpp_sendrecv.F90 [hunk 1]
```diff
@@ -149,7 +157,7 @@ subroutine test_sendrecv_2D_R4(npes,pe,root,out_unit)
 
     call mpp_sync() ! Needed ?
 
-    !! Verify that the data was corrrectly transmitted.
+    !! Verify that the data was correctly transmitted.
```
Comment (ngs333) [inline test_fms/mpp/test_mpp_sendrecv.F90:160]: Whats wrong with using three 
rs. What if we happen like the rolling sound? Isn't this a free country?

**Review by ngs333 — APPROVED** [review]

File: test_fms/mpp/test_mpp_sendrecv.F90 (hunk 1 above)
Comment (wrongkindofdoctor) [reply test_fms/mpp/test_mpp_sendrecv.F90:160]: You can still roll the r when you say it.

**Review by rem1776 — APPROVED** [review]

File: test_fms/mpp/test_mpp_sendrecv.F90 [hunk 2]
```diff
@@ -149,7 +157,7 @@ subroutine test_sendrecv_2D_R4(npes,pe,root,out_unit)
...
     if(ANY(pe == pelist(1:npes-1))) then
        do j = 1, DS 
           do i = 1, DS
```
Comment (mlee03) [inline test_fms/mpp/test_mpp_sendrecv.F90:163]: Will they all not equal since data is R4 and (i*10+j)'s are integers?

Comment (wrongkindofdoctor) [reply test_fms/mpp/test_mpp_sendrecv.F90:163]: Good catch! Presumably, the compiler was able to cast the integers to the appropriate types, allowing the tests to pass. I will print the some values to verify, and specify the values using a different method if needed

Comment (wrongkindofdoctor) [reply test_fms/mpp/test_mpp_sendrecv.F90:166]: The data appear to be correctly stored according to the data array type. However, I made the data value factors used to define the arrays in the R4 and R8 tests real values for insurance (can we ever REALLY trust the compiler O.o to do what we want?).

**Pull request merged**